registration/include/pcl/registration/icp.h

```diff
@@ -331,6 +332,54 @@ namespace pcl
                       PointCloudSource &output, 
                       const Matrix4 &transform);
   };
+
+  /** \brief @b SymmetricIterativeClosestPointWithNormals is a special case of
+    * IterativeClosestPoint, that uses a transformation estimated based on
+    * Symmetric Point to Plane distances by default.
+    *
+    * \author Matthew Cong
+    * \ingroup registration
+    */
+  template <typename PointSource, typename PointTarget, typename Scalar = float>
+  class SymmetricIterativeClosestPointWithNormals : public IterativeClosestPoint<PointSource, PointTarget, Scalar>
```

Comment:
I wonder if we need to introduce an entire new class for this. What do you think about adding an option "use symmetric objective" to the 'IterativeClosestPointWithNormals' class?

Reply:
Or more generally:

```Vector3 n = normal_weighing_function(source_it->getNormalVector3fMap(), target_it->getNormalVector3fMap(), weight_source, weight_target);```

We can take this as lambda input and simply provide some default lambdas who's behavior boils down to:
* weight_source = 1, weight_target = 0: obv
* weight_source = 1, weight_target = 1: symetric
* weight_source = 0, weight_target = 1:: point-to-plane weighted

Reply:
I've removed the new class and added the use symmetric objective option.

Reply:
Thanks for the suggestion, but could we please defer the weighting lambda to another PR? I think it could help with unifying weighting among the existing ICP implementations as well (e.g. point to plane vs. point to plane weighted which currently exist as separate classes). Thus, I feel that it's a bit out of scope for this feature PR.

Reply:
> Thanks for the suggestion, but could we please defer the weighting lambda to another PR?

Certainly. I've already seen way too many PRs that were sidetracked by endless refactoring opportunities :laughing: